src/main/scala/com/exasol/cloudetl/kinesis/KinesisShardsMetadataReader.scala

```diff
@@ -16,13 +16,14 @@ object KinesisShardsMetadataReader {
     val awsSessionToken = kinesisArgumentsReader.getAwsSessionTokenArgument
     val streamName = kinesisArgumentsReader.getStreamNameArgument
     val region = kinesisArgumentsReader.getRegionArgument
-    val kinesisMetadata = getKinesisShardsIds(
-      awsAccessKeyId,
-      awsSecretAccessKey,
-      awsSessionToken,
-      streamName,
-      region
-    )
```

Comment:
You can create again, `KinesisConsumerProperties`, using the first string from iterator.
For example, `val kinesisProperties = KinesisConsumerProperties(iterator.getString(0))` and remove arguments reader.

Reply:
done